vignettes/chevron_catalog.rmd

```diff
@@ -781,6 +781,98 @@ To display all possible grades even if they do not occur in the data, set the ar
 run(lbt14, syn_data, direction = "high", prune_0 = FALSE)
 ```
 
+
+### **Laboratory Test Shifts to `NCI-CTCAE` Grade 3-4 Post-Baseline (`LBT15`)**
+
+Please note that the following examples of summarizing lab grades using single and double grading approaches can be applied to other lab table templates beyond `LBT15`.
+
+#### **1. Laboratory Test Shifts to `NCI-CTCAE` Grade 3-4 Post-Baseline (Single Grading Approach)**
+
+In ADaM v1.1 or below, lab grade variables (`BTOXGR` and `ATOXGR`) in ADLB range from -4 to +4, with -/+ indicating the direction of lab abnormality. This is known as the single grading approach. 
+
+To summarize patients shifting to post-baseline grade 3-4 based on ADLB with the single grading approach, simply run the default `lbt15` template shown below.
+
+```{r}
+run(lbt15, syn_data)
+```
+
+#### **2. Laboratory Test Shifts to `NCI-CTCAE` Grade 3-4 Post-Baseline (Double Grading Approach)**
+
+In ADaM v1.2 or above, ADLB now has two sets of lab grade variables - the high-directional variables (`ATOXDSCH`/`ATOXGRH`/`BTOXGRH`) and the low-directional variables (`ATOXDSCL`/`BTOXGRL`/`ATOXGRL`), both ranging from 0 to 4. This is known as the double grading or the bi-directional approach, which is the default in `{admiral}`.
+
+To summarize patients shifting to post-baseline grade 3-4 based on ADLB with the double grading approach, redefining `ANRIND` and `BNRIND` is needed in the pre-processing step shown below.
+
+```{r}
+proc_data <- syn_data
+# Convert single grading variables to double ones for demo purpose only
+# as the original syn_data is using single grading approach
+proc_data$adlb <- proc_data$adlb %>%
+  mutate(
+    BTOXGRH = as.factor(
+      case_when(
+        BTOXGR %in% c("-4", "-3", "-2", "-1", "0") ~ "0",
+        TRUE ~ BTOXGR
+      )
+    ),
+    ATOXGRH = as.factor(
```

Comment:
Nice, but I think you could make use of `dunlin::reformat` that case:

```
rr <- rule("0" = c("-4", "-3", "-2", "-1", "0"))
...
mutate(y = reformat(.data$x, rr))
```

Reply:
Thank you so much @BFalquet for the tip! Great advice. I've implemented it in my latest commit. Please let me know if there's anything further I can improve. Thanks!